### PR TITLE
Solved #442- Proximity to POI

### PR DESCRIPTION
**Pull request by Aditi22Bansal**
## 🛠️ Related Issue

- Closes #[issue_number]
#442 
## 📝 Description

[Provide a concise description of the changes. Include the purpose and any relevant context or details.]

## 🔍 Type of PR

- [ ] 🐛 Bug fix
- [x] ✨ Feature enhancement
- [ ] 📚 Documentation update
- [ ] 🛑 Other (specify): _______________

## 📸 Screenshots / 🎥 Videos (if applicable)

[Attach screenshots or videos]

## ✅ Checklist

- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## ℹ️ Additional Context

[Include any additional context or information that might assist reviewers in understanding the changes]

### CODE REVIEW

**Comment by Aditi22Bansal** [comment]
@Akshat111111 
Kindly review, add labels and merge

File: Finacial Domain/Proximity to POI/POI.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+preprocessor = ColumnTransformer(
+    transformers=[
+        ('num', num_transformer, num_cols),
+        ('cat', cat_transformer, cat_cols)
+    ])
+
+# Define the full pipeline including polynomial features and the regression model
+pipeline = Pipeline(steps=[
+    ('preprocessor', preprocessor),
+    ('poly_features', PolynomialFeatures(degree=2, include_bias=False)),
```
Comment (Aditya-dom) [inline Finacial Domain/Proximity to POI/POI.py:49]: Remove 'degree=2' to be set in GridSearchCV

```suggestion
    ('poly_features', PolynomialFeatures( include_bias=False)),
```

File: Finacial Domain/Proximity to POI/POI.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    ('model', LinearRegression())
+])
+
+# Define parameter grid for GridSearchCV
+param_grid = {
+    'poly_features__degree': [1, 2, 3],
+    'model': [LinearRegression(), Ridge(), Lasso()],
+    'model__alpha': [0.1, 1.0, 10.0] if 'alpha' in grid_search.best_params_ else [0]
```
Comment (Aditya-dom) [inline Finacial Domain/Proximity to POI/POI.py:57]: Directly set 'alpha' for all models

```suggestion
    'model__alpha': [0.1, 1.0, 10.0]
```

**Review by Aditya-dom — CHANGES_REQUESTED** [review]

**Comment by Aditi22Bansal** [comment]
@Akshat111111 
@Aditya-dom 
Kindly review and merge

**Review by Aditya-dom — APPROVED** [review]
LMGT!!

**Comment by Aditi22Bansal** [comment]
@Aditya-dom 
Kindly add labels

**Comment by Aditi22Bansal** [comment]
@Aditya-dom 
For gssoc 24 leaderboard

**Review by Akshat111111 — COMMENTED** [review]
yeah, it works

**Comment by Akshat111111** [comment]
@Aditya-dom are you a part of gssoc ??

**Comment by Aditya-dom** [comment]
> @Aditya-dom are you a part of gssoc ??

Yup @Akshat111111 
I received an email assigning me to review the PRs as the deadlines are approaching.

**Comment by Akshat111111** [comment]
from what??

**Pull request merged**